cvat-core/src/server-proxy.ts

```diff
@@ -284,6 +285,19 @@ Axios.interceptors.response.use((response) => {
     return response;
 });
 
+axiosRetry(Axios, {
+    retryCondition: (error) => error.response && error.response.status === Axios.HttpStatusCode.TooManyRequests,
+    retryDelay: (retryCount, error) => {
+        const retryAfterValue = error.response.headers['retry-after'];
+
+        // Retry-After is allowed to be a number as well as a date.
+        // We're probably not going to use the latter option, though, so don't bother parsing it.
+        if (!retryAfterValue || !/^\d+$/.test(retryAfterValue)) return axiosRetry.exponentialDelay(retryCount, error);
+
+        return parseInt(retryAfterValue, 10) * 1000;
+    },
```

Comment:
If ``error.response.headers['retry-after']`` is supposed to be a number or not a number string:

```suggestion
    retryDelay: (retryCount: number, error: Error): number => {
        // Retry-After is allowed to be a number as well as a date.
        // We're probably not going to use the latter option, though, so don't bother parsing it.
        const retryTimeout = (+error.response.headers['retry-after']) * 1000;
        if (Number.isNaN(retryTimeout)) {
            return axiosRetry.exponentialDelay(retryCount, error);
        }
        return retryTimeout;
    },
```

Reply:
By "number" I really mean "non-negative decimal integer". See the [definition of `Retry-After`](https://www.rfc-editor.org/rfc/rfc9110#field.retry-after).

Blindly converting to a JavaScript number could yield undesirable values like negative numbers or infinity, so I'd rather not do that.

Reply:
Negative numbers are not covered by the code above.
For avoid infinity/floats we may use:

``Number.isInteger(retryTimeout) && retryTimeout > 0``

Instead of ``isNaN``

Reply:
> It is a little bit more pretty, than regex using

I don't know about pretty, but I find my version more straightforward. I verify that:

1. The header is present.
2. The header value follows the expected grammar.

Then I convert the value to the expected type.

Your version relies on JS's numeric conversion doing the right thing for values not matching the grammar, which isn't even true:

```console
> +'0xA'
10
```

Reply:
Why not true? You just converted HEX 0xA to decimal 10. 
And this is correct.

Reply:
![image](https://github.com/opencv/cvat/assets/40690378/68fa403b-23e9-4070-b681-4becd98129e1)

But we do not expect `+55` from the server, right? As well as ``0xA``..

Reply:
> Why not true? You just converted HEX 0xA to decimal 10.
> And this is correct.

It's not correct, because hexadecimal is not allowed by the grammar.

> Let's check with regex any data come from the server then...?

Sure? If it's amenable to such validation, then why not.

> ![image](https://private-user-images.githubusercontent.com/40690378/288036565-68fa403b-23e9-4070-b681-4becd98129e1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3ODUzMDksIm5iZiI6MTcwMTc4NTAwOSwicGF0aCI6Ii80MDY5MDM3OC8yODgwMzY1NjUtNjhmYTQwM2ItMjNlOS00MDcwLWI2ODEtNGJlY2Q5ODEyOWUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA1VDE0MDMyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFhMzAyMGQ1NGNlZjM0NzRkMDk2MGQxNmI1NDE1Njc4NWJmYmJhYjc2OGM3OWQxZWVmZmIzOTFmY2U1ODhkZDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jPirSLuyHi0omiWOuma7dgk7-s63jztz7T0zCSPcSEQ)

What's your point? Plus signs are not allowed by the grammar, so this string should be rejected - and that's what my implementation does.
 
> But we do not expect `+55` from the server, right? As well as `0xA`..

We don't - and that's why I reject such values.